`opa test --bench` panics on failing test

Tested with both v0.70.0 and current main. While I think `--bench` is meant to work with successful tests only, it seems like the error handling is borked.

**b.rego**
```rego
package b

import rego.v1

x := true

test_x if {
    not x
}
```

```
> opa test b.rego --bench
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10460281c]

goroutine 28 [running]:
testing.(*common).decorate(0x1400004fbe8?, {0x140003962d0, 0x29}, 0x1054439e0?)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/testing/testing.go:775 +0x9c
testing.(*common).logDepth(0x1400037fb88, {0x140003962d0, 0x29}, 0x3)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/testing/testing.go:1042 +0x150
testing.(*common).log(...)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/testing/testing.go:1011
testing.(*common).Fatal(0x1400037fb88, {0x1400004fee0?, 0x105670898?, 0x140003812d0?})
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/testing/testing.go:1082 +0x50
github.com/open-policy-agent/opa/tester.(*Runner).runBenchmark.func1(0x1400037fb88)
	/Users/anderseknert/git/opa/opa/tester/runner.go:588 +0x36c
testing.(*B).runN(0x1400037fb88, 0x1)
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/testing/benchmark.go:193 +0x128
testing.(*B).run1.func1()
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/testing/benchmark.go:215 +0x4c
created by testing.(*B).run1 in goroutine 26
	/opt/homebrew/Cellar/go/1.23.3/libexec/src/testing/benchmark.go:208 +0x90
```

If I change the test to succeed, the command works as expected.

--- a/cmd/test.go
+++ b/cmd/test.go
@@ -13,6 +13,7 @@ import (
 	"os/signal"
 	"strings"
 	"syscall"
+	"testing"
 	"time"
 
 	"github.com/fsnotify/fsnotify"
@@ -182,6 +183,11 @@ func runTests(ctx context.Context, txn storage.Transaction, runner *tester.Runne
 	var err error
 	var ch chan *tester.Result
 	if testParams.benchmark {
+		// Initialize testing package for benchmarking. This is needed to set default values for some flags that may
+		// otherwise be dereferenced on some code paths causing panics, as reported in:
+		// https://github.com/open-policy-agent/opa/issues/7205
+		testing.Init()
+
 		benchOpts := tester.BenchmarkOptions{
 			ReportAllocations: testParams.benchMem,
 		}
@@ -451,7 +457,7 @@ func init() {
 		Use:   "test <path> [path [...]]",
 		Short: "Execute Rego test cases",
 		Long: `Execute Rego test cases.
-	
+
 The 'test' command takes a file or directory path as input and executes all
 test cases discovered in matching files. Test cases are rules whose names have the prefix "test_".
 

--- a/cmd/test_test.go
+++ b/cmd/test_test.go
@@ -3474,3 +3474,27 @@ test_l if {
 		}
 	}
 }
+
+// Assert that a failing test doesn't cause a panic.
+// https://github.com/open-policy-agent/opa/issues/7205
+func TestTestBenchFailingTest(t *testing.T) {
+	files := map[string]string{
+		"test.rego": `package test
+			test_fail if false`,
+	}
+
+	test.WithTempFS(files, func(path string) {
+		fp := filepath.Join(path, "test.rego")
+		tp := newTestCommandParams()
+		tp.benchmark = true
+		tp.count = 1
+
+		exitCode, err := opaTest([]string{fp}, tp)
+		if exitCode == 0 {
+			t.Fatalf("Expected exit code 0, got %d", exitCode)
+		}
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+	})
+}

